### PR TITLE
Timezone support

### PR DESCRIPTION
**Pull request by bartero**
* interpret Accept-Timezone HTTP header
* implement DateTimeRangeParser, which accepts dates (and requirs them to be) in the ISO8601 format
* create a few time conversion helpers
* create helper `dateTimeToRequestTimezone` which is sensitive to the Accept-Timezone HTTP header

### CODE REVIEW

File: tests/functional/HelpersCest.php [hunk 1]
```diff
@@ -215,4 +216,62 @@ public function itAddsGroupArgumentsToRoutes(FunctionalTester $I)
...
+
+    public function itConvertsDateTimeObjectToRequestTimezone(FunctionalTester $I)
+    {
+        // 1st case: no header set
+        $I->sendGet(apiUrl('someapi'));
+
+        $dateTime = Carbon::parse('2021-05-02 12:43:31', 'America/New_York');
+        $dateTimeInRequestTimezone = dateTimeToRequestTimezone($dateTime);
+
+        $I->assertEquals($dateTime, $dateTimeInRequestTimezone);
```
Comment (AdrianSkierniewski) [inline tests/functional/HelpersCest.php]: should these two be equals here?

File: tests/unit/DateTimeRangeParserTest.php [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    {
+        $from = Carbon::parse('2021-05-02 0:43:31', 'Australia/Adelaide');
+        $to   = Carbon::parse('2021-05-01 23:43:31', 'America/New_York');
+
+        $parser = new DateTimeRangeParser('date');
+        $parser->parse(new Request(['date' => $from->toIso8601String() . "," . $to->toIso8601String()]));
+        $this->assertEquals('between', $parser->getOperation());
+        $result = $parser->getValue();
+
+        $this->assertEquals($from, $result[0]);
```
Comment (AdrianSkierniewski) [inline tests/unit/DateTimeRangeParserTest.php]: Same here. I don't understand why these two date-times are equals. Are they in different timezone right?

File: tests/functional/HelpersCest.php (hunk 1 above)
Comment (bartero) [reply tests/functional/HelpersCest.php]: These are equal in fact - they two point to the same moment in time.

File: tests/unit/DateTimeRangeParserTest.php (hunk 2 above)
Comment (bartero) [reply tests/unit/DateTimeRangeParserTest.php]: The same as above

**Pull request merged**